src/coreclr/jit/gentree.cpp

```diff
@@ -6298,13 +7098,15 @@ GenTree* Compiler::gtNewSIMDVectorZero(var_types simdType, CorInfoType simdBaseJ
 }
 #endif // FEATURE_SIMD
 
-GenTreeCall* Compiler::gtNewIndCallNode(GenTree* addr, var_types type, GenTreeCall::Use* args, const DebugInfo& di)
+GenTreeCall* Compiler::gtNewIndCallNode(GenTree* addr, var_types type, const DebugInfo& di)
 {
-    return gtNewCallNode(CT_INDIRECT, (CORINFO_METHOD_HANDLE)addr, type, args, di);
+    return gtNewCallNode(CT_INDIRECT, (CORINFO_METHOD_HANDLE)addr, type, di);
 }
 
-GenTreeCall* Compiler::gtNewCallNode(
-    gtCallTypes callType, CORINFO_METHOD_HANDLE callHnd, var_types type, GenTreeCall::Use* args, const DebugInfo& di)
+GenTreeCall* Compiler::gtNewCallNode(gtCallTypes           callType,
+                                     CORINFO_METHOD_HANDLE callHnd,
+                                     var_types             type,
+                                     const DebugInfo&      di)
```

Comment:
Yes, this should happen in callers now.
In general the arg list here was only used interestingly for JIT helpers, so I have folded them in as optional arguments in `gtNewHelperCallNode` that takes care of setting the flags.

For user calls, `impImportCall` will use `impPopCallArgs` and set the flags from the arguments after this. With that said, I have reviewed the sites that use `impPopCallArgs` and it looks like I have missed the flags in a couple of places... will fold it into `impPopCallArgs`.

I have also considered making `CallArgs` handle it when inserting arguments. It should be easy to get a back-pointer to the parent `GenTreeCall` given that it is stored inline. I think it would be nice not to be able to forget, but also it is conceptually wrong to do this after rationalization, so I am not totally sure if it makes sense to do always.

Reply:
FWIW, initially I wanted to replace the previous argument with `std::initializer_list`, but we cannot use STL :-(